### ISSUE TITLE
server freeze

### ISSUE THREAD

**Issue by PhanaticD** [PaperMC/Paper#896]
with build 1220 two of my servers froze right after startup with this same bit showing up in the server thread when doing a jstack on it https://pastebin.com/26nQfecM

**Comment by PhanaticD** [comment]
also should be worth noting that it was only 2/17 servers, and they did work fine after killing and starting them up again

**Comment by stephan-gh** [comment]
That's weird. This shouldn't happen anymore because https://github.com/PaperMC/Paper/commit/5a3fb946d8b66c21894c7b6e72227e3512f3c009 was added as a workaround for it. For some reason the workaround isn't working correctly for all of your servers...

Do you have Essentials installed on the other 15 servers as well or just on the two that are failing now?

**Comment by PhanaticD** [comment]
they all have essentials

**Comment by PhanaticD** [comment]
happened again with a different plugin, again with logging related. I can provide the complete thread dump if needed. (still on build 1220)
```
"Server thread" #19 prio=5 os_prio=0 tid=0x000069e29678d000 nid=0x7c8f runnable [0x000069dd82205000]
   java.lang.Thread.State: RUNNABLE
        at java.util.logging.Logger.getEffectiveLoggerBundle(Logger.java:2184)
        at java.util.logging.Logger.doLog(Logger.java:758)
        at java.util.logging.Logger.log(Logger.java:788)
        at java.util.logging.Logger.info(Logger.java:1490)
        at com.jroossien.portalguns.PortalGuns.log(PortalGuns.java:162)
        at com.jroossien.portalguns.PortalGuns.onEnable(PortalGuns.java:112)
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:316)
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:404)
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:386)
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:335)
        at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:442)
        at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:403)
        at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:341)
        at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:292)
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:616)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by stephan-gh** [comment]
Does this happen every time you start these servers or just sometimes? It's hard to tell what's causing these issues without being able to reproduce them reliably. It doesn't happen for me when I run Essentials or that plugin.

The only reason why this would happen is that the workaround I added isn't working in some cases, which should only happen if something loads the logger before the plugin is even loaded...

**Comment by PhanaticD** [comment]
it only happens sometimes, more often not. I had a minigame server that only froze once despite restarting about ever half hour all day yesterday

**Comment by PhanaticD** [comment]
still happens on build 1226

happened 3 times in a row when started from a crontab, when i started it manually it worked, not sure if that is just a coincidence though

**Comment by electronicboy** [comment]
Did you ever get around to testing the branch that @kashike created?

**Comment by PhanaticD** [comment]
yes it fixed the issue for me

**Comment by zachbr** [comment]
Merged into master w/ 3d3ee1d9b443610164ff17a897d65209bf654e78
Builds 1229+
